Character_Builder/models.py

```diff
@@ -35,14 +35,53 @@
 # As this is in the character builder folder, this will focus on
 # the character information 
 
+# Automatically creates CharacterIDs
+def newCharacterID():
+    previousCharacter = Character.objects.last()
+    if not previousCharacter is None:
+        newID = previousCharacter.characterID + 1
+    else:
+        newID = 0
+
+    return newID
+
+# Automatically creates AbilityScoreSetIDs
+def newAbilityScoreSetID():
+    previousAbilityScoreSet = AbilityScoreSet.objects.last()
+    if not previousAbilityScoreSet is None:
+        newID = previousAbilityScoreSet.abilityScoreSetID + 1
+    else:
+        newID = 0
+
+    return newID
+
+# finds a default user
+def defaultUser():
+    default = User.objects.first()
+
+    if default is None:
+        default = User.objects.create_user('defaultUser', password='djangoproject')
+
+    return default
+
+# Automatically creates abililtyScoreSetIDs
+def newAbilityScoreSetID():
+    previousCharacter = Character.objects.last()
+    if not previousCharacter is None:
+        newID = previousCharacter.abilityScoreSetID + 1
+    else:
+        newID = 0
+
+    return newID
+
 # This class is dynamic, the level, xp, hp, alignment, and (rarely) size may change
 class Character(models.Model):
-    user = models.ForeignKey(User, on_delete=models.CASCADE)
-    characterID = models.IntegerField(unique=True) # Note that Django has a built-in primary key
+    user = models.ForeignKey(User, on_delete=models.CASCADE, default=defaultUser)
+    characterID = models.IntegerField(unique=True, default=newCharacterID) # Note that Django has a built-in primary key
```

Comment:
Yeah, I probably should refactor this into

characterID = models.AutoField(primary_key=True)